### ISSUE TITLE
How to access features and target of final processed dataset?

### ISSUE THREAD

**Issue by anubhavkamal** [microsoft/BatteryML#28]

**Comment by agiamason** [comment]
Apologies for the delayed response. 
**For features:**
When you execute `batteryml run configs/baselines/sklearn/variance_model/matr_1.yaml ./workspace/test --train --eval`, the data will be cached in the `BatteryML/cache` folder. The cached files will have names in the format `battery_cache_xx.pkl`.

To load the data from these cached pickle files, you can use the following keys:

- `data['dataset']`: This contains the features.
- `data['raw_data']`: This contains the processed raw data.

The naming of the cache files is based on a hash of the config settings in the `config.yaml` file. We use the hash_string function in `BatteryML/batteryml/pipeline.py` for this.

**For predictions:**
As for the predictions, they are stored in `BatteryML/workspaces/<your model>/<dataset>/predictions_seed_n_xx.pkl`. Here, <your model> and <dataset> should be replaced with the name of your model and dataset, respectively.

I hope this clarifies your question. Let me know if you have any further queries.

**Comment by SaraaSameer** [comment]
Hello, I have a similar inquiry. I have downloaded the raw and preprocessed data for MATR. I have trained the model for both the variance and full models, and I have pickle files stored in the cache. However, when I try to retrieve the features for these models, I am only getting a data object. How can I extract the cycle and summary features for both the train and test datasets? 
Have attached the screenshot for reference

![Image](https://github.com/user-attachments/assets/72178735-f36e-45d7-b646-df4637e451b2)

**Comment by agiamason** [comment]
Hi @SaraaSameer, I apologize for any inconvenience caused. I hope my response can be helpful to you.
The `data['dataset']` is an instance of the `DataBundle` class. If you need to access the features of the training data, you can use `data['dataset'].train_data.feature`. To access the labels of the training data, use `data['dataset'].train_data.label`. Similarly, to obtain the features of the test data, you can use `data['dataset'].test_data.feature`,etc.

**Comment by SaraaSameer** [comment]
Thanks @agiamason for clarifying. I have extracted BatteryML data for the MATR variance model, thanks to your response above, but the CSV file lacks feature names. I am unable to locate the feature names in the code. Is there a way to retrieve them?
Please see the snippet of CSV for reference.

![Image](https://github.com/user-attachments/assets/17936124-5ba0-4430-81cc-2b0362a3338a)

**Comment by agiamason** [comment]
Hi @SaraaSameer ,  I understand the problem you are facing, but since some features such as QV curve are 1000-dimensional, the feature column names are not marked. I am afraid you need to look for it in the feature code file now. For example, the Variance model has only one feature, which can be found in `BatteryML/batteryml/feature
/variance_model.py` and is called `'Variance'`. I guess the screenshot you took may be the full model? There are 9 features, and you can see the list of feature names in `BatteryML/batteryml/feature
/full_model.py`
